### ISSUE TITLE
Implementation of the uncertainty loss

### ISSUE THREAD

**Issue by Aonnghus** [anger-man/cascaded-null-space-learning#1]
Hi,
First of all, thank you for your work! I've been reading your paper and found it really interesting. It is a very elegant way to estimate the uncertainty of a network. 
I checked out this repo a bit, and I've got some questions regarding your implementation of the uncertainty loss.

From method `forward` of class `uncMSE`:
```
unc = torch.clip(unc,self.eps,1e7)
tmp = torch.square(y_true-y_pred)/(2*torch.square(unc))
tmp = torch.where(tmp<=2,tmp,2+torch.log(torch.clip(tmp-2+1,self.eps,1e7)))
return torch.mean(tmp + .5*torch.log(1.+2*unc*np.pi))
```

I do not understand why you apply a logarithm to the uncertainty output when its values are superior to 2 (line 3 of the previous code), and I don't grasp the utility of `2+torch.log` and `tmp-2+1`. I've got the same question regarding the last line of the above code (`log(1.+2*unc...)`).

Thank you for your help!

**Comment by anger-man** [comment]
Hi,
Thank your very much for your comment.
I made the modifications  for stability reasons during the first few iterations. Line 3 simply says:
**if the outcome of the fraction in line 2 is greater than 2, than the logarithms should be applied to** _tmp_, **otherwise none.**
This helps to avoid large gradient steps in the beginning, especially when torch.square(unc) is close to zero.

You can also try to simply leave out line 3, should not make a big difference. 
Best regards, Christoph

**Comment by Aonnghus** [comment]
Thanks for this very fast answer!
